src/config.rs

```diff
@@ -0,0 +1,54 @@
+use crate::cli;
+use clap::Parser;
+use config::{Config as LibConfig, File};
+use serde::Deserialize;
+use std::{net::SocketAddr, path::PathBuf};
+
+#[derive(Deserialize, Clone, Default)]
+#[serde(default)]
+pub struct Config {
+    pub server: ServerConfiguration,
+    pub uml_creator: UMLCreatorConfiguration,
+}
+
+#[derive(Deserialize, Clone)]
+#[serde(default)]
+pub struct ServerConfiguration {
+    pub addr: SocketAddr,
+}
+
+impl Default for ServerConfiguration {
+    fn default() -> Self {
+        Self {
+            addr: ("0.0.0.0:8043").parse().expect("should be valid url"),
+        }
+    }
+}
+
+#[derive(Deserialize, Clone)]
+#[serde(default)]
+pub struct UMLCreatorConfiguration {
```

Comment:
```suggestion
pub struct UmlCreatorConfiguration {
```
https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case